src/reporters/udp_sender.js

```diff
@@ -107,8 +127,34 @@ export default class UDPSender {
         return {err: false, numSpans: numSpans};
     }
 
+    _convertBatchToThriftMessage() {
+        let spanMessages = [];
+        for (let i = 0; i < this._batch.spans.length; i++) {
+            let span = this._batch.spans[i];
+            spanMessages.push(new this._thrift.Span(span))
+        }
+
+        let tagMessages = [];
+        for (let j = 0; j < this._batch.process.tags.length; j++) {
+            let tag = this._batch.process.tags[j];
+            tagMessages.push(new this._thrift.Tag(tag));
+        }
+
+        return new this._thrift.Agent.emitBatch.ArgumentsMessage({
+            version: 1,
+            id: 0,
+            body: {batch: new this._thrift.Batch({
+                    process: new this._thrift.Process({
```

Comment:
you need to assign _something_ to `process: ` every time, no argument. But why does that _something_ needs to be reconstructed every time, given that the arguments to the construction never change?